src/core/issues-notes.js

```diff
@@ -14,158 +14,150 @@ import { pub } from "core/pubsubhub";
-            )
-          );
-          $inno.removeAttr("title");
-        }
-        $tit.addClass("marker");
-        $div.append($tit);
-        $inno.replaceWith($div);
-        var body = $inno.removeClass(report.type).removeAttr("data-number");
-        if (ghIssue && !body.text().trim()) {
-          body = ghIssue.body_html;
-        }
-        $div.append(body);
-        const level = $tit.parents("section").length + 2;
-        $tit.attr("aria-level", level);
       }
-      pub(report.type, report);
-    });
-    if ($(".issue").length) {
-      if ($("#issue-summary"))
-        $("#issue-summary").append($issueSummary.contents());
-    } else if ($("#issue-summary").length) {
-      pub("warn", "Using issue summary (#issue-summary) but no issues found.");
-      $("#issue-summary").remove();
+      $tit.find("span").text(text);
+      if (report.title) {
+        $tit.append(
+          $("<span style='text-transform: none'>: " + report.title + "</span>")
+        );
+        $inno.removeAttr("title");
+      }
+      $tit.addClass("marker");
+      $div.append($tit);
+      $inno.replaceWith($div);
+      var body = $inno.removeClass(report.type).removeAttr("data-number");
+      if (ghIssue && !body.text().trim()) {
+        body = ghIssue.body_html;
+      }
+      $div.append(body);
+      const level = $tit.parents("section").length + 2;
+      $tit.attr("aria-level", level);
     }
+    pub(report.type, report);
+  });
+  if ($(".issue").length) {
+    if ($("#issue-summary"))
+      $("#issue-summary").append($issueSummary.contents());
+  } else if ($("#issue-summary").length) {
+    pub("warn", "Using issue summary (#issue-summary) but no issues found.");
+    $("#issue-summary").remove();
+  }
+}
+export async function run(conf) {
+  const $ins = $(".issue, .note, .warning, .ednote");
+  const ghIssues = {};
+  const { issueBase } = conf;
+  if (!$ins.length) {
+    return; // nothing to do.
   }
-  var $ins = $(".issue, .note, .warning, .ednote"),
-    ghIssues = {},
-    issueBase = conf.issueBase;
-  if ($ins.length) {
-    if (conf.githubAPI) {
-      ghFetch(conf.githubAPI)
-        .then(function (json) {
-          issueBase = issueBase || json.html_url + "/issues/";
-          return fetchIndex(json.issues_url, {
-            // Get back HTML content instead of markdown
-            // See: https://developer.github.com/v3/media/
-            Accept: "application/vnd.github.v3.html+json",
-          });
-        })
-        .then(function (issues) {
-          issues.forEach(function (issue) {
-            ghIssues[issue.number] = issue;
-          });
-          handleIssues($ins, ghIssues, issueBase);
-          cb();
-        }).catch(err => {
-          pub("error", err.message);
-          handleIssues($ins, ghIssues, issueBase);
-          cb();
-        });
-    } else {
-      handleIssues($ins, ghIssues, issueBase);
-      cb();
+  if (conf.githubAPI) {
```

Comment:
I'm still not super happy with this part... I think we should ONLY got to network if:

```
if (conf.githubAPI && document.querySelector("issue[data-number]") ){

}
```

Reply:
Another significant improvement would be to _only_ fetch the issues in the spec... we are downloading all the issues for the repo... that's hugely wasteful.